ksqldb-tools/src/test/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommandTest.java

```diff
@@ -0,0 +1,340 @@
+import io.confluent.ksql.tools.migrations.MigrationConfig;
+import io.confluent.ksql.tools.migrations.util.MetadataUtil;
+import io.confluent.ksql.tools.migrations.util.MetadataUtil.MigrationState;
+import io.confluent.ksql.tools.migrations.util.MigrationsDirectoryUtil;
+import io.vertx.core.json.JsonArray;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.Charset;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+
+public class ApplyMigrationCommandTest {
+
+  private static final SingleCommand<ApplyMigrationCommand> PARSER =
+      SingleCommand.singleCommand(ApplyMigrationCommand.class);
+
+  private static final String MIGRATIONS_TABLE = "migrations_table";
+  private static final String MIGRATIONS_STREAM = "migrations_stream";
+  private static final String NAME = "FOO";
+  private static final String COMMAND = "CREATE STREAM FOO (A STRING) WITH (KAFKA_TOPIC='FOO', PARTITIONS=1, VALUE_FORMAT='DELIMITED');";
+
+  @Rule
+  public TemporaryFolder folder = new TemporaryFolder();
+
+  @Mock
+  private MigrationConfig config;
+  @Mock
+  private Client ksqlClient;
+  @Mock
+  private BatchedQueryResult versionQueryResult;
+  @Mock
+  private BatchedQueryResult infoQueryResult;
+  @Mock
+  private CompletableFuture<Void> insertResult;
+  @Mock
+  private CompletableFuture<ExecuteStatementResult> statementResult;
+
+  private String migrationsDir;
+  private ApplyMigrationCommand command;
+
+  @Before
+  public void setUp() throws ExecutionException, InterruptedException {
+    when(config.getString(MigrationConfig.KSQL_MIGRATIONS_TABLE_NAME)).thenReturn(MIGRATIONS_TABLE);
+    when(config.getString(MigrationConfig.KSQL_MIGRATIONS_STREAM_NAME)).thenReturn(MIGRATIONS_STREAM);
+
+    when(ksqlClient.insertInto(any(), any())).thenReturn(insertResult);
+    when(ksqlClient.executeStatement(any())).thenReturn(statementResult);
+    when(ksqlClient.executeQuery("SELECT VERSION FROM " + MIGRATIONS_TABLE + " WHERE version_key = 'CURRENT';"))
+        .thenReturn(versionQueryResult);
+    when(ksqlClient.executeQuery("SELECT checksum, previous, state FROM " + MIGRATIONS_TABLE + " WHERE version_key = '1';"))
+        .thenReturn(infoQueryResult);
+    when(insertResult.get()).thenReturn(null);
+    when(statementResult.get()).thenReturn(null);
+
+    migrationsDir = folder.getRoot().getPath();
+  }
+
+  @Test
+  public void shouldApplyFirstMigration()
+      throws IOException, ExecutionException, InterruptedException {
```

Comment:
nit: let's just use `throws Exception` so we don't have to keep these lists up to date.